### PR TITLE
CDPT-2079 AJAX search fix & refactor

### PR DESCRIPTION
**Pull request by EarthlingDavey**
This PR includes a refactor of archive pages.

AJAX filter/search results are pulled in via ajax.

The underlying WP_Query for the AJAX results is the same as  the WP_Query used for loading the first page of results (without AJAX).

Some bugs have been ironed out and unnecessary code has been removed.

There are still a lot of potential improvements to be made, e.g. 

- remove `get-news-rest-api.php` and re-write functions that rely on it.
- remove `pagination.php` and modify templates that use it, as a 2nd WP_Query should not be run just for pagination.

^ having removed the file `get-blog-rest-api.php` and refactoring pagination on `page_news.php` and `page_blog.php` shows that the above improvements are feasible.

### CODE REVIEW

File: public/app/themes/clarity/src/globals/js/ajax-utils.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  }
+
+  // Load an instance of the AjaxTemplating class.
+  const t = new AjaxTemplating(templateName);
+
+  // Render the html for each post.
+  const resultsHtml = posts.map((props) => t.renderHtml(props));
+
+  // Append the html to the content section.
+  $("#content").append(resultsHtml);
```
[the remark this answers is not in the document]
Comment (EarthlingDavey) [reply public/app/themes/clarity/src/globals/js/ajax-utils.js:91]: I think this is OK as I do see a way that an attacker could inject malicious code into `resultsHtml`.

Would you mind offering a second opinion please @wilson1000 ?

[the remark this answers is not in the document]
Comment (EarthlingDavey) [reply public/app/themes/clarity/src/globals/js/ajax-utils.js:91]: I think this is OK as I do see a way that an attacker could inject malicious code into `resultsHtml`.

Would you mind offering a second opinion please @wilson1000 ?

[the remark this answers is not in the document]
Comment (wilson1000) [reply public/app/themes/clarity/src/globals/js/ajax-utils.js:91]: Can we test this? It looks good 😄

File: public/app/themes/clarity/functions.php [hunk 2]
```diff
@@ -120,6 +119,9 @@
 new MOJ\Intranet\WPOffloadMedia();
 new MOJ\Intranet\WPElasticPress();
 
+$search = new MOJ\Intranet\Search();
+$search->hooks();
```
Comment (wilson1000) [inline public/app/themes/clarity/functions.php:123]: Do we have to notate it like this? Can we follow the pattern of include and or do we need other $search variables in the `functions.php` script?

**Review by wilson1000 — COMMENTED** [review]
Hi @EarthlingDavey , it looks like you ventured down a rabbit hole here 😄 
Can you walk me through the update tomorrow? It would be good to get your view of the improvements you've made. Thank you!

**Review by wilson1000 — APPROVED** [review]
👍

**Pull request merged**